app/models/concerns/labware.rb

```diff
@@ -18,6 +18,7 @@ def prefix
   def generate_barcode
     return if barcode.present?
 
-    update(barcode: "TRAC-#{self.class.prefix}-#{id}")
+    self.barcode = "TRAC-#{self.class.prefix}-#{id}"
+    save(validate: false)
```

Comment:
This probably deserves a comment as it is unusual

Reply:
I was thinking about this and think I will try to instead stub out the Tube in the tests so this validation can still be performed

Reply:
Addressed this by injecting my own object on creation so that any validation performed during object creation can be distinguished from validation performed in the factory